tests/api/test_metrics.py

```diff
@@ -0,0 +1,83 @@
+# Copyright (c) 2024 Snowflake Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from snowflake.cli.api.metrics import CLIMetrics
+
+
+def test_metrics_no_counters():
+    # given
+    expected = {}
+    metrics = CLIMetrics()
+
+    # when
+
+    # then
+    assert metrics.counters == expected
+    assert metrics.get_counter("counter1") is None
+
+
+def test_metrics_set_one_counter():
+    # given
+    expected = {"counter1": 1}
+    metrics = CLIMetrics()
+
+    # when
+    metrics.set_counter("counter1", 1)
+
+    # then
+    assert metrics.counters == expected
+    assert metrics.get_counter("counter1") == expected["counter1"]
+
+
+def test_metrics_add_new_counter():
+    # given
+    expected = {"counter1": 2}
```

Comment:
nit: expected is not needed here, unless it's a complex construction.. in that case, it can be inline at the end:
```
assert metrics.counters == {'counter1': 2}
```